### PR TITLE
Hide password in the index printed via `--emit-index-annotation`

### PR DESCRIPTION
**Pull request by ChannyClaus**
## Summary
resolves https://github.com/astral-sh/uv/issues/3106

## Test Plan
added a simple test where the password provided in `UV_INDEX_URL` is hidden in the output as expected.

### CODE REVIEW

File: crates/uv-resolver/src/resolution.rs [hunk 1]
```diff
@@ -746,7 +746,12 @@ impl std::fmt::Display for DisplayResolutionGraph<'_> {
             // `# from https://pypi.org/simple`).
             if self.include_index_annotation {
                 if let Some(index) = node.index() {
-                    writeln!(f, "{}", format!("    # from {index}").green())?;
+                    let mut url = index.url().clone();
+                    if url.password().is_some() {
+                        // Hide the password.
+                        let _ = url.set_password(Some("****"));
+                    }
+                    writeln!(f, "{}", format!("    # from {url}").green())?;
```
Comment (charliermarsh) [inline crates/uv-resolver/src/resolution.rs:754]: Maybe we should just omit the username and password entirely -- what do you think?

Comment (ChannyClaus) [reply crates/uv-resolver/src/resolution.rs:754]: yup, that is cleaner. making the changes -

**Review by charliermarsh — APPROVED** [review]
Thanks!

File: crates/uv-resolver/src/resolution.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-resolver/src/resolution.rs:750]: Just moved it out to a method. Thanks!

**Pull request merged**